extensions/src/main/java/org/pentaho/platform/web/http/api/resources/MantleResource.java

```diff
@@ -49,32 +52,34 @@
 public class MantleResource {
 
   private final Log log = LogFactory.getLog( MantleResource.class );
-  private static final List<String> whiteListedHosts = new ArrayList<>();
+  private static List<String> whiteListedHosts = new ArrayList<>();
 
+  protected File propFile;
   @GET
   @Path( "/getDeeplinkAllowedHosts" )
   @Produces( { APPLICATION_JSON } )
-  @StatusCodes( { @ResponseCode( code = 200, condition = "Successfully sent result." ),
+  @StatusCodes( { @ResponseCode( code = 200, condition = "Successfully obtained allowed host list." ),
     @ResponseCode( code = 500, condition = "Failed to get data." ) } )
-  public List<String> getDeeplinkAllowedHosts() {
-    return getListHosts();
+  public Response getDeeplinkAllowedHosts() {
+    try {
+      return Response.ok(getWrappedResponse(), MediaType.APPLICATION_JSON).build();
+    }catch (Exception e){
+      return Response.serverError().build();
+    }
+  }
+
+  HostsWrapper getWrappedResponse() {
+    return new HostsWrapper(getListHosts());
   }
 
   private List<String> getListHosts() {
     if ( whiteListedHosts.isEmpty() ) {
 
-      File propFile = new File(
-        PentahoSystem.getApplicationContext().getSolutionRootPath() + "/system/deeplink_allowed_urls.properties" );
+      getPropFile();
```

Comment:
I'm confused about this line.
`getPropFile();`

Before the values of file deeplink_allowed_urls.properties were used inside this function. But with the proposed changes, this file isn't used.

And why call a function that the results are never used?